server/sockets/cohort.js

```diff
@@ -237,218 +262,139 @@ const updateCohortHeaderStatus = (socket, cohortClientLocks) => {
+      }
+    });
 
-        docs.forEach(doc => {
-          const { type } = doc;
-          const listId = doc._id.toString();
+    listIdsUserWasRemovedFrom.forEach(async listId => {
+      const data = { listId, performer, userId };
+      try {
+        await emitRemoveListViewer(io, CohortActionTypes.REMOVE_MEMBER_SUCCESS)(
+          data
+        );
+      } catch {
+        // Ignore errors
+      }
+    });
 
-          if (isViewer(doc, userId)) {
-            listIdsUserRemained.push(listId);
-          } else if (type === ListType.SHARED) {
-            listIdsUserWasRemovedFrom.push(listId);
-          }
+    listIdsUserRemained.foreach(listId => {
+      io.sockets
+        .to(listChannel(listId))
+        .emit(ListActionTypes.MEMBER_UPDATE_SUCCESS, {
+          isGuest: true,
+          listId,
+          userId
         });
-
-        if (listIdsUserWasRemovedFrom.length > 0) {
-          if (dashboardClients.has(userId)) {
-            const { socketId } = dashboardClients.get(userId);
-
-            io.sockets
-              .to(socketId)
-              .emit(ListActionTypes.REMOVE_BY_IDS, listIdsUserWasRemovedFrom);
-          }
-
-          listIdsUserWasRemovedFrom.forEach(listId => {
-            io.sockets
-              .to(listChannel(listId))
-              .emit(ListActionTypes.REMOVE_MEMBER_SUCCESS, {
-                listId,
-                userId
-              });
-
-            if (listClients.has(userId)) {
-              const { socketId, viewId } = listClients.get(userId);
-              const isCohortMember = checkIfArrayContainsUserId(
-                cohortMemberIds,
-                userId
-              );
-
-              if (viewId === listId) {
-                io.sockets
-                  .to(socketId)
-                  .emit(ListActionTypes.REMOVED_BY_SOMEONE, {
-                    cohortId,
-                    isCohortMember,
-                    listId
-                  });
-              }
-            }
-          });
-        }
-
-        if (listIdsUserRemained.length > 0) {
-          listIdsUserRemained.forEach(listId => {
-            if (listClients.has(userId)) {
-              const { socketId, viewId } = listClients.get(userId);
-
-              if (viewId === listId) {
-                io.sockets
-                  .to(socketId)
-                  .emit(ListActionTypes.MEMBER_UPDATE_SUCCESS, {
-                    isCurrentUserUpdated: true,
-                    isGuest: true,
-                    listId,
-                    userId
-                  });
-              }
-            }
-
-            io.sockets
-              .to(listChannel(listId))
-              .emit(ListActionTypes.MEMBER_UPDATE_SUCCESS, {
-                isCurrentUserUpdated: false,
-                isGuest: true,
-                listId,
-                userId
-              });
-          });
-        }
-      }
     });
+  } catch {
+    // Ignore error
+  }
+
+  return Promise.resolve();
```

Comment:
please also check here is we still need this

Reply:
In sockets' method where we get data from DB - yes, eg. we often get socketIds from DB